Update To QT6

TheForce172 opened the pull request:
Starts the process of updating us to QT6.

~*Crash on close on windows (GLSync Error)~ Fixed
~*Segmentation Fault on Launch in Linux VM~  Not reproducble
~*Not a crash but this required fully rewriting the CMake file so more changes may me needed to that and possibly to workflows.~  Being handled in separate PR by taylor

Also this will require us to drop Windows 7 support as QT6 Does not support it.

TheForce172 committed:
Enable Qt6

TheForce172 committed:
Fix Main Window For DPI Scaling

TheForce172 committed:
Add back in RCC

TheForce172 committed:
Fix Deprecations

TheForce172 committed:
Build Fix and remove illegal chars

TheForce172 committed:
Fix more illegal Chars

TheForce172 committed:
Fix Several Crashes on close

Move shutdown to run before Qt destroys the opengl context

disconnect visibility changes as it accidently runs after windows destruction due to QT weirdness

TheForce172 committed:
Fix Merge Error

TheForce172 committed:
Qt6 Changes

TheForce172 committed:
Not entirely Sure why but VS studio debug needs this to run

TheForce172 committed:
Build qthelp for QT6

MjnMixael committed:
fix qtfred help

TheForce172 committed:
Try to fix briefing editor

TheForce172 committed:
supress dpi context warning

TheForce172 committed:
Try and disable console on non dev machines

notimaginative committed:
bump macOS deployment target to 10.15

notimaginative committed:
add support for precompiled Qt libs

notimaginative committed:
fix executable type for Windows/Mac

notimaginative committed:
fix finding of qhelpgenerator on Mac/Linux

notimaginative committed:
place help files in proper location on Mac

notimaginative committed:
add Mac app bundle info for qtFRED

notimaginative committed:
fix binary identification in AppImage script

With the new QtFRED help changes it gets the filename wrong, including the
names of multiple files by mistake. This tweaks the find command to ignore
those other files.

notimaginative committed:
Mjn's help cache fix

notimaginative committed:
add warning to not distribute builds using system Qt install

notimaginative committed:
switch to universal deployment setup

Use Qt deployment cmake script for all platforms. This uses the platform
specific method where possible (windeployqt/macdeployqt) and a generic
deployment script on other platforms.

This also uses a crazy RPATH hack on Linux, which is necessary for it to
find the prebuilt Qt libs and deploy them. Qt does not appear to allow
passing tool options for the generic deployment script so we can't tell
the relavent cmake functions where to look for our libraries even though
they support such functionality.

notimaginative committed:
add custom fonts for Linux and fix platform check

notimaginative committed:
remove AppImage hack for broken Wayland in Qt5

notimaginative committed:
stop install of things we don't need

notimaginative committed:
fix cmake policy warning

notimaginative committed:
stop installation of runtime libs (KNet does it)

notimaginative committed:
remove broken imported config mapping

TheForce172 committed:
Reapply #7450

BMagnu commented on qtfred/src/FredApplication.cpp:
```diff
@@ -1,86 +1,86 @@
+	connect(this, &FredApplication::initializeComplete, [this]() { _initializeEmitted = true; });
+
+	// Run our shutdown code after closing last window, but before application quits.
+	//connect(qApp, &QApplication::lastWindowClosed, this, &FredApplication::shutdown);
```
Is it fine that we aren't actually running the shutdown code? That seems less than ideal

TheForce172 replied:
We do, shutdown is run in the close event on line 2198 of Fredview.cpp, We were having a conflict here with both FSO and QT trying to destroy the opengl context after the FredView window had closed where QT had already destroyed the context. This will need to be refactored if we add support for multiple fred windows which is seem was planed at one point?

BMagnu replied:
Gotcha, that's fine for now then

BMagnu reviewed (commented):
One question, otherwise LGTM conditional on it being rebased once #7562 is in, with Qt tests / builds reenabled.

notimaginative committed:
enable QtFRED by default

Note that Win32 is not supported by Qt6 and so QtFRED is forcibly disabled
on 32-bit Windows.

notimaginative committed:
update prebuilt ref to Qt6 version

notimaginative committed:
use prebuilt Qt libs for CI

notimaginative committed:
fix prebuilt issue when used outside of /lib

notimaginative committed:
bump prebuilt version

TheForce172 committed:
Fix more deprecations that somehow managed to slip by.

TheForce172 committed:
One more depreciation

notimaginative committed:
add prebuilt fallback if system Qt not available

TheForce172 committed:
Clang Fix

notimaginative committed:
set QtFRED option before toolchain is included

notimaginative committed:
fix lib fallback

notimaginative committed:
fix compile error

notimaginative committed:
address standard c++ lib issues

Disable static libstdc++/libc++ by default now that we've moved to using a
base of Ubuntu 22.04. Using a static c++ library also poses a possible issue
with our prebuilt Qt6 libs since they are dynamically linked.

Disable automatic use of libc++ when static c++ library is enabled. The
options can and should be used independently.

Disable using libc++ for clang by default since it conflicts with the prebuilt
Qt6 libs as well as the system Qt6 libs on most Linux distros. An message has
been added to QtFRED config to error/warn if this is likely to be a problem.

TheForce172 committed:
Merge pull request #8 from notimaginative/qtfred_at_night

fix PR checks and add QtFRED to nightly builds

TheForce172 committed:
3 Clang Fixes

TheForce172 committed:
One more
